design/authenticators/authn_jwt/token_schema.md

```diff
@@ -0,0 +1,180 @@
+
+| **Term**         | **Description**                                              |
+| ---------------- | ------------------------------------------------------------ |
+| JWT Claims       | claims are pieces of information asserted about a subject    |
+| Host Annotation  | Defining of check that we should do on authentication request for host. In JWT authentication its a claim check. |
+| Mandatory Claims | JWT Claims that host must validate their existance and value |
+| Claims Mapping   | Mapping between claim name and name of host annotation       |
+| Token Schema     | Combination of mandatory claims and claims mapping           |
+
+## Useful Links
+[//]: # "Add links that may be useful for the reader"
+
+| **Name**   | **Link**                                                     |
+| ---------- | ------------------------------------------------------------ |
+| JWT Claims | https://auth0.com/docs/tokens/json-web-tokens/json-web-token-claims |
+
+## Requirments
+[//]: # "Elaborate on the issue you are writing a solution for"
+
+1. Required fields
+   1. The user will have the ability to declare in the authenticator policy which claims must be in host annotations. And therefore also in the JWT token.
+   2. Upon authentication request the authenticator MUST validate those claims - existence and content 
+2. Mapping 
+   1. The user should have the ability to map a given "technical" claim name to a more familiar / business one - i.e, in git-lab - instead of mention in the policy "sub" claim, he will have the ability to mention "job"
+
+## Solution
+[//]: # "Elaborate on the solution you are suggesting in this page. Address the functional requirements and the non functional requirements that this solution is addressing. If there are a few options considered for the solution, mention them and explain why the actual solution was chosen over them. Add an execution plan when relevant. It doesn't have to be a full breakdown of the feature, but just a recommendation to how the solution should be approached."
+
+User with define the token schema variables in authenticator policy. They will be checked in the JWT authentication request.
+
+### User Interface
+
+[//]: # "Describe user interface (including command structure, inputs/outputs, etc where relevant)"
+
+#### Authenticator Policy
+
+Two new variables will be added to the authenticator policy:
+
+* mandatory_claims
+* claims_mapping
+
+```yaml
+    - !policy
+    id: conjur/authn-jwt/VendorY
+    body:
+    - !webservice
+    
+    - !variable
+      id: mandatory_claims
+      
+    - !variable
+      id: claims_mapping
+```
+
+#### Variable Values Example
+
+| Name             | Value                   | Description                                      |
+| ---------------- | ----------------------- | ------------------------------------------------ |
+| mandatory_claims | branch, job, aud        | List of claims as they appear in host annotation |
```

Comment:
Are "branch, job, aud" only an example ? Do you have a full list of values for mandatory claims ?

Reply:
These are examples